yield_plugin/src/yield_plugin.cpp

```diff
@@ -37,72 +37,381 @@ using oss = std::ostringstream;
+                                            MobilityResponseCB mobility_response_publisher)
+    : wm_(wm), config_(config), plugin_discovery_publisher_(plugin_discovery_publisher), mobility_response_publisher_(mobility_response_publisher)
   {
     plugin_discovery_msg_.name = "YieldPlugin";
     plugin_discovery_msg_.versionId = "v1.0";
     plugin_discovery_msg_.available = true;
     plugin_discovery_msg_.activated = false;
     plugin_discovery_msg_.type = cav_msgs::Plugin::TACTICAL;
-    plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";
+    plugin_discovery_msg_.capability = "tactical_plan/plan_trajectory";   
   }
 
-  bool YieldPlugin::onSpin()
+  void YieldPlugin::lookupECEFtoMapTransform()
+  {
+    tf2_listener_.reset(new tf2_ros::TransformListener(tf2_buffer_));
+    tf2_buffer_.setUsingDedicatedThread(true);
+    try
+    {
+      tf_ = tf2_buffer_.lookupTransform("earth", "map", ros::Time(0), ros::Duration(20.0)); //save to local copy of transform 20 sec timeout
+    }
+    catch (const tf2::TransformException &ex)
+    {
+      ROS_WARN("%s", ex.what());
+    }
+  }
+
+  bool YieldPlugin::onSpin() 
   {
     plugin_discovery_publisher_(plugin_discovery_msg_);
     return true;
   }
 
+  void YieldPlugin::set_lanechange_status_publisher(const ros::Publisher& publisher)
+  {
+    lanechange_status_pub_ = publisher;
+  }
+
+  std::vector<lanelet::BasicPoint2d> YieldPlugin::detect_trajectories_intersection(std::vector<lanelet::BasicPoint2d> trajectory1, std::vector<lanelet::BasicPoint2d> trajectory2) const
+  {
+    // the vectors are converted to linestrings so their intersection point can be calculated
+
+    // A linestring is a Curve with linear interpolation between Points.
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> traj1;
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> traj2;
+    for (auto tpp1:trajectory1)
+    {
+      boost::geometry::append(traj1, tpp1);
+    }
+
+    for (auto tpp2:trajectory2)
+    {
+      boost::geometry::append(traj2, tpp2);
+    }
+
+    std::vector<lanelet::BasicPoint2d> intersection_points;
+    boost::geometry::intersection(traj1, traj2, intersection_points);
+
+    return intersection_points;
+  }
+
+  std::vector<lanelet::BasicPoint2d> YieldPlugin::detect_trajectories_intersection2(std::vector<lanelet::BasicPoint2d> self_trajectory, std::vector<lanelet::BasicPoint2d> incoming_trajectory) const
+  {
+    std::vector<lanelet::BasicPoint2d> intersection_points;
+    boost::geometry::model::linestring<lanelet::BasicPoint2d> self_traj;
+    for (auto tpp:self_trajectory)
+    {
+      boost::geometry::append(self_traj, tpp);
+    }
+    // distance to consider trajectories colliding (chosen based on lane width and vehicle size)
+    double acceptable_distance = 2.0;
+    for (size_t i=0; i<incoming_trajectory.size(); i++)
+    {
+      double res = boost::geometry::distance(incoming_trajectory[i], self_traj);
```

Comment:
So this computing the distance between that point and the closest point on our current path? That might work, but I'm still not 100% sure it of it's soundness without considering the time-aspect of the conflict/collision warning. 

If the proposed lane change (from the mobility request) comes from a vehicle behind us (but in an adjacent lane) such that the last point of their trajectory aligns basically with the first few points of our current trajectory (but 6s after we'd be there), are we going to flag that as a conflict and slow down (thus causing an actual lane change conflict)?

Reply:
this function is computing the distance between the point and the linestring, which would be the closest point in the line. And yes, I believe it would detect the conflict in the lane change scenario.
About time, since we are checking the times a few times, so it should be okay.
Also, I had the same concern when developing this, and then I realized the time is more important if it was car following where trajectories just overlap, not in the case of an incoming lane change where a point can be found as the first point of intersection. Here, it is more about speed adjustment reaching that point.